app/src/main/java/com/sdp/movemeet/utility/ActivityCache.java

```diff
@@ -0,0 +1,56 @@
+package com.sdp.movemeet.utility;
+
+
+import android.content.pm.PackageManager;
+import android.os.Environment;
+import android.util.Log;
+import android.widget.Toast;
+
+import androidx.appcompat.app.AppCompatActivity;
+
+import com.sdp.movemeet.Activity.Activity;
+
+import org.apache.commons.lang3.SerializationUtils;
+
+import java.io.BufferedReader;
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.FileNotFoundException;
+import java.io.FileOutputStream;
+import java.io.IOException;
+import java.io.InputStreamReader;
+import java.io.ObjectInput;
+import java.io.ObjectInputStream;
+import java.io.ObjectOutput;
+import java.io.ObjectOutputStream;
+import java.util.ArrayList;
+
+import static android.content.Context.MODE_PRIVATE;
+
+public class ActivityCache extends AppCompatActivity {
+
+    public void saveActivitiesInCache(ArrayList<Activity> activities) {
+        ObjectOutput out;
+        try {
+            File outFile = new File(Environment.getExternalStorageDirectory(), "cached_activities");
```

Comment:
Simple, efficient, nice! Good if local cache works this way in that easy manner!

Reply:
Thank you, but unfortunately it isn't working yet :(
I think the problem comes from the permissions rather than the way the data is stored locally.